### PR TITLE
Update actions to fix report build

### PR DESCRIPTION
**Pull request by danfarrow**
Adds action to merge any commits to `trunk` into `report`

Uses [devmasx/merge-branch](https://github.com/devmasx/merge-branch)

### CODE REVIEW

File: .github/workflows/merge-trunk-to-report.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+on:
+  push:
+    branches:
+      # Run on pushes to trunk
+      - 'trunk'
+jobs:
+  merge-branch:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
```
Comment (ryelle) [inline .github/workflows/merge-trunk-to-report.yml]: In the other action, we use `actions/checkout@v2` - is there a reason to use `master` here?

File: .github/workflows/build-report.yml [hunk 2]
```diff
@@ -2,21 +2,22 @@ name: Build Report
...
+      - report
 
-  schedule:
-    # Run every day at 9am UTC.
-    # * is a special character in YAML so you have to quote this string
-    - cron:  '0 9 * * *'
+   schedule:
+     # Run every day at 9am UTC.
+     # * is a special character in YAML so you have to quote this string
+     - cron:  '0 9 * * *'
```
Comment (ryelle) [inline .github/workflows/build-report.yml]: I think these added spaces might break the yaml parsing, it uses indentation for nesting so it's really sensitive.

File: .github/workflows/merge-trunk-to-report.yml (hunk 1 above)
Comment (ryelle) [reply .github/workflows/merge-trunk-to-report.yml]: it looks like checkout deprecated the master branch, so this should be `v2`.

**Review by ryelle — COMMENTED** [review]
The merge trunk action looks good, I think that should work how we want 👍🏻 

The only downside here is that actions don't trigger other actions, so the action merging trunk to report will not trigger the report rebuild. Maybe that's fine, since the schedule will still run daily - what do you think?

File: .github/workflows/merge-trunk-to-report.yml (hunk 1 above)
Comment (danfarrow) [reply .github/workflows/merge-trunk-to-report.yml]: re: _is there a reason to use master here?_ no I think I just copied & pasted it like that 😁  - I can change to use v2

Comment (danfarrow) [reply .github/workflows/merge-trunk-to-report.yml]: Oh I see you fixed it already, thanks!

File: .github/workflows/build-report.yml (hunk 2 above)
Comment (danfarrow) [reply .github/workflows/build-report.yml]: oops - unintentional!

**Comment by danfarrow** [comment]
re: _Maybe that's fine, since the schedule will still run daily - what do you think?_  that’s the reason I was considering merging into one action, but then I wasn’t sure what the trigger should be. But yes I think that’s 💯% fine for now

File: .github/workflows/merge-trunk-to-report.yml (hunk 1 above)
Comment (ryelle) [reply .github/workflows/merge-trunk-to-report.yml]: I didn't fix anything, it's still master in `merge-trunk-to-report.yml`

Comment (danfarrow) [reply .github/workflows/merge-trunk-to-report.yml]: 😅 fixed now! Also probably notified some random guy whose GitHub username is `v2` (I’m sure it’s not the first time for him!)

File: .github/workflows/build-report.yml (hunk 2 above)
Comment (danfarrow) [reply .github/workflows/build-report.yml]: Now reverted the whitespace!

**Review by ryelle — APPROVED** [review]
Let's try it!

**Comment by danfarrow** [comment]
Merged! `merge-trunk-to-report` seems to have succeeded. Will wait to see if `build-report` does.
Should `/public/style.css` & `/public/wp-admin.html` now be deleted from trunk?

**Comment by ryelle** [comment]
🎉  Looks like [it worked!](https://github.com/WordPress/css-audit/actions/runs/905802065) I see the commits from trunk, and a new report was generated. Great job!

> Should /public/style.css & /public/wp-admin.html now be deleted from trunk?

Yeah, that make sense since that report will be outdated.

**Comment by danfarrow** [comment]
I am buzzing with excitement here! That’s awesome!

**Pull request merged**